engine/class_modules/sc_rogue.cpp

```diff
@@ -4117,6 +4123,14 @@ struct stealth_t : public rogue_spell_t
 
     p()->buffs.stealth->trigger();
     trigger_master_of_shadows();
+
+    // trigger_master_of_shadows() doesn't trigger out of combat, must be applied manually
+    if ( precombat_seconds && !p()->in_combat )
```

Comment:
I think either trigger_master_of_shadows() needs to be in an else case here, or this logic needs to be moved into the trigger_master_of_shadows() function itself with a precombat_seconds optional parameter or something. 

Think it's a little unclear to someone browsing this code as to why we are triggering MoS twice here or doing so manually when there is already a function here.